core/src/main/java/org/ldaptive/pool/AbstractConnectionPool.java

```diff
@@ -988,7 +988,14 @@ public PooledConnectionStatistics getPooledConnectionStatistics()
     @Override
     public boolean equals(final Object o)
     {
-      return LdapUtils.areEqual(this, o);
+      if (o == this) {
```

Comment:
Since it's a proxy, I wanted equality to pass through to the underlying object. Note that equality isn't defined on the underlying object.